### DISCUSSION TITLE
API e2e testing: How to check for a specific cookie in the response

### DISCUSSION THREAD

**Discussion by NorseJedi** [pactumjs/pactum#316] [Q&A] [answered]
I'm trying to check if an API-call sets a refresh_token cookie, but I can't seem to figure out how. The value of the cookie is a JWT so I can't check the actual value, I just want to check if it gets set.

I have tried:
```typescript
.expectCookiesLike('refresh_token', 'string')
```
...which fails with: `Cookie doesn't have value 'string' at '$.refresh_token' but found 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOjEsImVtYWlsIjoibm9AbWFpbC5jb20iLCJpYXQiOjE3MDg3Nzk4MTgsImV4cCI6MTcwOTk4OTQxOH0.mBI3Cgvle1jNRD7f1wgijRk6Lsc-fHjg_0jXwLSn9eI'`

I have tried:
```typescript
.expectCookiesLike('refresh_token', String)`
```
...which fails with: `Cookie doesn't have type 'function' at '$.refresh_token' but found 'string'`

After struggling for ages to figure out the correct syntax for `expectJsonSchema` I even tried:
```typescript
.expectCookiesLike({'refresh_token': {
    'type': 'string'
}})
```
...which also fails with: `Cookie doesn't have type 'object' at '$.refresh_token' but found 'string'`

Basically it seems that whatever i try to specify other than `'string'` it fails because it finds `'string'`, but when I specify `'string'` it fails because it finds the actual token (which is a string).

I'm fairly green when it comes to automated testing, which is what I'm trying to change, but I suspect I'm missing something painfully obvious and that I can't see the forest for the trees... A little help, please? :)

(Using pactum 3.6.0 with NestJS 10.3.3 and Nx 18.0.4)

**Comment by NorseJedi** [comment, marked as answer]
Ok, so I kept hammering away at this even though I'd given up while waiting and hoping for someone to come to my rescue here, but I found a solution with RegExp.
The docs do not mention that this is possible for `expectCookiesLike()` but it does describe it for `expectJsonLike()` so I figured "Like" and "Like", maybe they're handled the same way under the hood, so I gave it one more shot, and it worked. I'm posting here in case someone else googles for ages with little luck like I did, so hopefully they'll find this :)

So the solution to this specific case would be something like this:
```typescript
.expectCookiesLike('refresh_token', /[A-Za-z0-9\-_.]{128,}/)
```
This should match any JWT with atleast 128 characters(*). The smallest I was able to generate on jwt.io was 84, but that was with no actual data. The ones in my API are currently 135ish but they'll get bigger before I'm done. Specifying an upper limit here is probably unnecessary, atleast for my use.

**Reply by ASaiAnudeep** [reply]
Yes, internally it uses [expectJsonLike](https://pactumjs.github.io/api/assertions/expectJsonLike.html). May it makes sense to build something like `expectCookieMatch` which will be similar to [expectJsonMatch](https://pactumjs.github.io/api/assertions/expectJsonMatch.html)

**Reply by NorseJedi** [reply]
That may be, but I'd also suggest updating the docs to show that regular expressions can be used this way. The docs only refer to lightcookie which also doesn't seem to document any possibility of using regular expressions. Adding an example that illustrates how it can be done would also be nice. Feel free to use my solution as an example :)
